### DISCUSSION TITLE
Configure clipboard permission

### DISCUSSION THREAD

**Discussion by tobihans** [seleniumbase/SeleniumBase#1295] [Q&A] [answered]
Hello! I'm new to Selenium testing.  
I have an app where some actions copy text to clipboard. My problem with the tests is that I execute some JavaScript from SeleniumBase to get clipboard content. This trigger permission asking on Chrome and block. If I'm in `--gui` mode, I can enable it by clicking otherwise it will timeout and my test will fail. I searched for different ways I can enable clipboard permissions and I found an answer [here](https://stackoverflow.com/questions/64107578/accept-clipboard-permission-request-in-chrome-using-selenium). So I'd like to know how I can set custom options for the browser or if there is a better way to give permissions.

Below is the code I wrote for the test.

```python3
    def test_copy_as_csv_action(self):
        self.goto(self.get_server_url())
        self.reload()
        checkbox = self.find_elements('input[type="checkbox"]')[0]
        checkbox.click()
        self.click('#actions .dropdown button')
        self.click('#copy-as-csv')
        self.assert_element('.success-popup')
        clipboard_content = self.execute_async_script(
            'let c = await navigator.clipboard.readText(); arguments[0](c)')
        print(clipboard_content)
        self.wait(5)
```
Thanks.

**Comment by mdmintz** [comment, marked as answer]
Hi @tobihans,
For setting custom Chromium options, you can add the following to your ``pytest`` run command:

```bash
--chromium-arg="ARG=N,ARG2"  # (Set Chromium args, ","-separated, no spaces.)
```

If you're just trying to copy/paste some text, you might not even need to use the clipboard. You can grab the text of an area by using: ``self.get_text(selector)``. Then, there are several ways of pasting the text in. For a pure paste, you can do ``self.add_text(selector, text)`` / ``self.send_keys(selector, text)``, or if replacing existing text, use ``self.type(selector, text)``, which clears the text field before putting in the text.

When running on a headless Linux display, you can add ``--xvfb`` to ``pytest`` so that you use a virtual display, which can get around limitations when running in headless mode. That will allow you to run tests like in gui mode.

**Reply by tobihans** [reply]
Thanks for your answer @mdmintz. But is there a programmatic way to do it? For instance, being able to configure it from a class that subclass BaseCase.

**Reply by mdmintz** [reply]
For the command-line options that you always want activate, all you have to do is throw those in a ``pytest.ini`` file on the ``addopts`` row.
Eg. https://github.com/seleniumbase/SeleniumBase/blob/master/pytest.ini
